### ISSUE TITLE
Model obj mismatch with camera.xml

### ISSUE THREAD

**Issue by HrsPythonix** [facebookresearch/EyefulTower#5]
Hello, I tried to import your camera.xml and mesh.obj in metashape, I found that they did not seem to align with each other on some scene. Is there some information I missed like shift when importing model?

Riverview:
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/b43f6a95-0c10-492c-bed8-08e60d29b60a)
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/4755ac7d-1b3b-4377-b1aa-3a9d7f434e92)

kitchen:
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/71983517-d9a1-476f-aba6-ac764e313d6a)
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/dcad93c8-8a1d-4377-8c08-5ed8637af524)

**Comment by cr333** [comment]
Both `cameras.xml` and `mesh.obj` were exported from the same project and are aligned correctly:

![image](https://github.com/facebookresearch/EyefulTower/assets/8134692/29ca472b-6cd4-4b85-a291-afa599dd69b2)

The orientation of your mesh looks correct, but the camera rigs seem to be upside-down or rotated.

One noticeable difference is that our project uses multi-camera rigs, i.e. all cameras are grouped together (only 48" images" as far as Metashape is concerned, while your project shows 1056 images).

Can you please share the exact steps you took to import the files into a new Metashape project?

**Comment by HrsPythonix** [comment]
@cr333 
1. Workflow -> Add Folders -> Select images-jpeg-2k in apartment
2. Choose Add all images to one chunk
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/3cb1c170-21c2-4128-91ce-5323de5565f4)
3. Modify the camera.xml as in https://github.com/facebookresearch/EyefulTower/issues/4
```
Modifying all <resolution width="5784" height="8660"/> to <resolution width="1368" height="2048"/>
and
scale all <f>, <cx>, <cy> by 4.228515625 (8660 / 2048). Without updating pixel_width and pixel_width
```
4. File -> import cameras and select modified camera.xml

Modified camera.xml of apartment is pasted in this post, as github does not allow uploading xml file, I rename the file by adding a .log, plz get rid of it before you use it.
[cameras.xml.log](https://github.com/facebookresearch/EyefulTower/files/13420990/cameras.xml.log)

**Comment by HrsPythonix** [comment]
I tried use Multi-camera system in riverview scene
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/56567adf-c8a8-4488-8240-5d8d778d82bd)

and import unchanged camera.xml, but it seems not right yet
![image](https://github.com/facebookresearch/EyefulTower/assets/16586180/e242c30d-624a-4e2e-910b-f41c0954ab8c)

**Comment by cr333** [comment]
I'm afraid this looks like a Metashape issue to me: the `cameras.xml` file is not imported correctly.

This is where the cameras should be relative to the bounding region (orthographic view):
![image](https://github.com/facebookresearch/EyefulTower/assets/8134692/2bd55485-d961-449b-9b03-a22074ba62ae)

And this is where the cameras are after exporting them to `cameras.xml` and importing them into a new project using your instructions:
![image](https://github.com/facebookresearch/EyefulTower/assets/8134692/e585fc47-742c-4ab5-97d1-c58b1c4762bf)

The camera poses are translated and rotated, it seems.

One older post on the Metashape forum suggest that Metashape may apply a translation on import:
> Currently when the cameras are imported from the bundler.out format the additional translation is applied to the imported data to bring the chunk origin to the point related to the average camera positions.
— [Alexey Pasumansky](https://www.agisoft.com/forum/index.php?topic=5567.msg27240#msg27240), Agisoft Technical Support

A [second post](https://www.agisoft.com/forum/index.php?topic=6270.0) goes into more detail, with code, on how to correctly import mesh and cameras from bundler. While this seems highly related, it is also stated that this has been fixed in version 1.3.2, so I'm sure how applicable this is.

Unfortunately, I don't have the time to debug this any further at this point. I'm attaching a version of our Metashape project for `riverview` with the correct camera poses to help you debug this further: [riverview-final-cameras-only.zip](https://github.com/facebookresearch/EyefulTower/files/13504746/riverview-final-cameras-only.zip)

Please take this up with Metashape's support or forum, and report back when there is a solution. I would be very grateful for that.

**Comment by HrsPythonix** [comment]
> I'm afraid this looks like a Metashape issue to me: the `cameras.xml` file is not imported correctly.
> 
> This is where the cameras should be relative to the bounding region (orthographic view): ![image](https://private-user-images.githubusercontent.com/8134692/286713740-2bd55485-d961-449b-9b03-a22074ba62ae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzMDk1NzksIm5iZiI6MTcwMTMwOTI3OSwicGF0aCI6Ii84MTM0NjkyLzI4NjcxMzc0MC0yYmQ1NTQ4NS1kOTYxLTQ0OWItOWIwMy1hMjIwNzRiYTYyYWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMzBUMDE1NDM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGFlZmUxMWY1ZTY5NDA0YTc3NTA4ZDVmMjhhMzVkNDJhODkwOGU2ZWUzNGI1NmYzN2YzOTRkNzBiMWY5ZTQ4MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.x6t83Cd6enPZ0WJ7Yoh2dZ0hMDLamXlxQ6d1sYqeMDE)
> 
> And this is where the cameras are after exporting them to `cameras.xml` and importing them into a new project using your instructions: ![image](https://private-user-images.githubusercontent.com/8134692/286713956-e585fc47-742c-4ab5-97d1-c58b1c4762bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDEzMDk1NzksIm5iZiI6MTcwMTMwOTI3OSwicGF0aCI6Ii84MTM0NjkyLzI4NjcxMzk1Ni1lNTg1ZmM0Ny03NDJjLTRhYjUtOTdkMS1jNThiMWM0NzYyYmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTEzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzExMzBUMDE1NDM5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YWRiNDZiODBmZDk5ZTE5MDlkMDYyOGVhMzBjNTQyMjgxZDVmMTYwOGE2YzE5ZDFiZjcxOTNiNTlkMDQwZmJkOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0lQ4P5ythdqp0JVXNF_9MAsDLiBv_Y9rXpU7Pws3dYE)
> 
> The camera poses are translated and rotated, it seems.
> 
> One older post on the Metashape forum suggest that Metashape may apply a translation on import:
> 
> > Currently when the cameras are imported from the bundler.out format the additional translation is applied to the imported data to bring the chunk origin to the point related to the average camera positions.
> > — [Alexey Pasumansky](https://www.agisoft.com/forum/index.php?topic=5567.msg27240#msg27240), Agisoft Technical Support
> 
> A [second post](https://www.agisoft.com/forum/index.php?topic=6270.0) goes into more detail, with code, on how to correctly import mesh and cameras from bundler. While this seems highly related, it is also stated that this has been fixed in version 1.3.2, so I'm sure how applicable this is.
> 
> Unfortunately, I don't have the time to debug this any further at this point. I'm attaching a version of our Metashape project for `riverview` with the correct camera poses to help you debug this further: [riverview-final-cameras-only.zip](https://github.com/facebookresearch/EyefulTower/files/13504746/riverview-final-cameras-only.zip)
> 
> Please take this up with Metashape's support or forum, and report back when there is a solution. I would be very grateful for that.

Wondering if you can provide me more scene like [riverview-final-cameras-only.zip](https://github.com/facebookresearch/EyefulTower/files/13504746/riverview-final-cameras-only.zip) you pasted scene here to help with further debugging, like apartment and kitchen, which I have already run metashape process.
